plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/AbstractTestParquetReader.java

```diff
@@ -2164,4 +2171,38 @@ private static SqlDate intToSqlDate(Integer input)
         }
         return new SqlDate(input);
     }
+
+    private static CustomParquetWriter getParquetWriter(MessageType schema, File outputFile)
+            throws IOException
+    {
+        Path path = new Path(outputFile.toURI().toString());
+        return new CustomParquetWriter(path, schema, false, CompressionCodecName.SNAPPY);
+    }
+
+    @Test
+    public void testDoubleByteStreamSplitSingleColumn()
+            throws IOException
+    {
+        List<String> columnNames = ImmutableList.of("byte_stream_split_sample");
+        MessageType parquetSchema = parseMessageType("message byte_stream_split_test { optional double byte_stream_split_sample; }");
+        try (ParquetTester.TempFile tempFile = new ParquetTester.TempFile("test", "parquet")) {
+            ContiguousSet<Long> epochMillisValues = ContiguousSet.create(Range.closedOpen((long) -1_000, (long) 1_000), DiscreteDomain.longs());
+            epochMillisValues.stream().forEach(System.out::println);
+            List<List<String>> stringListOfLists = epochMillisValues.stream().map(n -> Arrays.asList(String.valueOf(n * 1.0))).collect(Collectors.toList());
+            List<Double> doubleList = epochMillisValues.stream().map(n -> n * 1.0).collect(Collectors.toList());
+
+            CustomParquetWriter writer = getParquetWriter(parquetSchema, tempFile.getFile());
```

Comment:
I don't see how we ensured BYTE_STREAM_SPLIT encoding in this writer. I think it needs use of `org.apache.parquet.hadoop.ParquetWriter.Builder#withByteStreamSplitEncoding`.
Could you generate a parquet file with BYTE_STREAM_SPLIT encoding for a float and a double using some other tool, add that to test resources and write a test on reading that ? It would simplify testing.

Reply:
One way you can try is to temporarily update `ParquetWriter#initColumnWriters` to use `.withByteStreamSplitEncoding(true)`. Then use `set session hive.parquet_optimized_writer_enabled=true;` and then write a parquet file with a double and a float column type. That should give us a file with this encoding which we can use for tests.